pkg/flow/import_test.go

```diff
@@ -0,0 +1,362 @@
+package flow_test
+
+import (
+	"context"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/grafana/agent/pkg/flow"
+	"github.com/grafana/agent/pkg/flow/internal/testcomponents"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/service"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/tools/txtar"
+
+	_ "github.com/grafana/agent/component/module/string"
+)
+
+// updateFile is used to change the content of a file used by a module
+type updateFile struct {
+	name         string   // name of the file which should be updated (module | nested_module)
+	updateConfig []string // new module config which should be used
+}
+
+func TestImport(t *testing.T) {
```

Comment:
Nice! This is much easier to follow than before. I have some light feedback though:

As it is now, you have to modify two files (the txtar file and this file) to add a new test case, which can be slightly burdensome. Have you considered an approach similar to what the converters test use, where the test automatically loads files in a directory, and uses individual files (or pairs of files) as the test cases? 

If you move to multiple files, you can use some conventions for defining new test cases. For example:


```txtar
This file tests import passthrough module. 

-- main.river -- 
testcomponents.count "inc" {
	frequency = "10ms"
	max = 10
}

import.file "testImport" {
	filename = "module.river"
}

testImport.a "cc" {
	input = testcomponents.count.inc.count
}

testcomponents.summation "sum" {
	input = testImport.a.cc.output
}

-- module.river -- 
declare "a" {
	argument "input" {}

	testcomponents.passthrough "pt" {
		input = argument.input.value
		lag = "1ms"
	}

	export "output" {
		value = testcomponents.passthrough.pt.output
	}
}

-- update/module.river -- 
declare "a" {
	argument "input" {}

	export "output" {
		value = -argument.input.value
	}
}
```

In the example above, the following conventions are used:

* The description of the txtar explains the test and what it's doing. 
* The primary configuration file to load is always called `main.river`. 
* Other files in the same directory are unpacked on disk in a temporary directory (so `main.river` and `module.river` are unpacked together) where the module is ran.
* Files in the `update/` directory overwrite any existing files on demand; such that when you want to update files, `update/module.river` overwrites `module.river`. 

I don't know if these conventions make the most sense for these tests (I only came up with the first example that came to mind), but it just serves as an example for some improvements that can be made. The main point of my feedback is to try to find a way to create new test cases in a single place without having to touch multiple files.

Reply:
tha's a good suggestion. I initially used one file to reduce redundancy and allow modules to be reused and concatenated to form new usecases but I agree that it's not the best when it comes to add new tests and we also end up with a big file that is hard to navigate. Your approach makes it much easier to follow, I will make the changes